packages/core/src/test/awsService/ec2/model.test.ts

```diff
@@ -144,6 +147,22 @@ describe('Ec2ConnectClient', function () {
             sinon.assert.calledWith(sendCommandStub, testSelection.instanceId, 'AWS-RunShellScript')
             sinon.restore()
         })
+
+        it('avoids writing the keys to any telemetry metrics', async function () {
+            sinon.stub(SsmClient.prototype, 'sendCommandAndWait')
+
+            const testSelection = {
+                instanceId: 'test-id',
+                region: 'test-region',
+            }
+            const keyPath = path.join((await createTestWorkspaceFolder()).uri.path, 'keys')
+            const keys = await SshKeyPair.getSshKeyPair(keyPath, 30000)
+            await client.sendSshKeyToInstance(testSelection, keys, 'test-user')
+            const privKey = await fs.readFileText(keys.getPrivateKeyPath())
+            assertNoTelemetryMatch(privKey)
```

Comment:
Is there some other metric we can positively assert, before this assert? That ensures that this doesn't pass merely because of a race. Alternatively, sleep for 50ms :(